tests/time_integration/bdf_time_derivative_01.cc

```diff
@@ -0,0 +1,107 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#include <iostream>
+#include <numeric>
+
+#include <deal.II/lac/la_vector.h>
+
+#include <exadg/time_integration/bdf_constants.h>
+#include <exadg/utilities/enum_utilities.h>
+
+// Check compute_bdf_time_derivative() from quantities and times
+
+using namespace ExaDG;
+using VectorType = dealii::LinearAlgebra::Vector<double>;
+
+enum class Shape
+{
+  Constant,
+  Linear
+};
+
+void
+test(unsigned int const order, Shape shape)
+{
+  std::cout << "Check derivative of " << Utilities::enum_to_string(shape) << " shape" << std::endl;
+
+  // construct quantites
+  std::vector<VectorType> quantities_np_values;
+
+  double target_slope = -99.0;
+  if(shape == Shape::Linear)
+  {
+    // construct quantities with slope -1
+    target_slope = -1.0;
+    for(unsigned int i = 0; i < order + 1; ++i)
+    {
+      VectorType vec(1);
+      vec[0] = i;
+      quantities_np_values.push_back(vec);
+    }
+  }
+  else if(shape == Shape::Constant)
+  {
+    // construct quantities with slope 0
+    target_slope = 0.0;
+    for(unsigned int i = 0; i < order + 1; ++i)
+    {
+      VectorType vec(1);
+      vec[0] = 1.0;
+      quantities_np_values.push_back(vec);
+    }
+  }
+
+  // transfrom to vector of pointers
+  std::vector<VectorType const *> quantities_np(quantities_np_values.size());
+  std::transform(quantities_np_values.begin(),
+                 quantities_np_values.end(),
+                 quantities_np.begin(),
+                 [](VectorType const & t) { return &t; });
+
+
+  // construct times which are in reverse order in realistic applications
+  std::vector<double> times_np(order + 1);
+  std::iota(times_np.begin(), times_np.end(), 0.0);
+  std::sort(times_np.begin(), times_np.end(), [](auto const a, auto const b) { return a > b; });
+
+
+  // compute temporal derivative
+  VectorType derivative(1);
+  derivative[0] = -99.0;
+  compute_bdf_time_derivative(derivative, quantities_np, times_np);
+
+  // check if computed slope equals target slope
+  assert(std::abs(derivative[0] - target_slope) < 1e-12);
```

Comment:
Since `target_slope` is not used in the computations above, I would argue this test does not ensure that the implementation is correct, but only that the same results are obtained as when writing the test.